utils/build-script-impl

```diff
@@ -2265,17 +2440,17 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
                 if [[ -z "${INSTALL_LLDB}" ]] ; then
                     continue
                 fi
-                case "$(uname -s)" in
-                    Linux)
+                case ${host} in
+                    linux-*)
                         ;;
-                    FreeBSD)
+                    freebsd-*)
                         ;;
-                    CYGWIN_NT-10.0)
+                    cygwin-*)
                         ;;
-                    Darwin)
+                    macosx-*)
                         set_lldb_build_mode
                         pushd ${LLDB_SOURCE_DIR}
-                        xcodebuild -target toolchain -configuration ${LLDB_BUILD_MODE} install ${lldb_xcodebuild_options[@]} DSTROOT="${INSTALL_DESTDIR}" LLDB_TOOLCHAIN_PREFIX="${TOOLCHAIN_PREFIX}"
+                        xcodebuild -target toolchain -configuration ${LLDB_BUILD_MODE} install ${lldb_xcodebuild_options[@]} DSTROOT="${host_install_destdir}"
```

Comment:
> It seems to not always append a path separator

This part can be fixed.

> In order to build a clean toolchain in both situations (cross-compiling and OSX-only), I had to:

What were the build and host machines?

The problem here is that the compiler wants to install itself into `<prefix>/bin/swift`, but we want it in `<toolchain>/usr/bin/swift`, so we set `prefix=<toolchain>/usr`.  But LLDB needs to install files into `<toolchain>/System`, so it has to know the path to the toolchain directory.  LLDB's Xcode build system also has native knowledge of toolchains, so I'd be very hesitant to avoid using `LLDB_TOOLCHAIN_PREFIX`.